packages/ethers-cf-purelocal/index.ts

```diff
@@ -0,0 +1,75 @@
+import { ethers } from "ethers";
+import VM from "ethereumjs-vm";
+
+const vm = new VM({
+  enableHomestead: true,
+  activatePrecompiles: true
+});
+export class PureEvmProvider extends ethers.providers.JsonRpcProvider {

```

Comment:
Intuitively the thing to extend seems like `Contract` (so you can access the `abi` property)